### DISCUSSION TITLE
More "Rust is not hard!" comparisons

### DISCUSSION THREAD

**Discussion by dbanty** [dbanty/dylananthony.com#205] [Ideas]
I want to compare the difficulty of implementing different types of software projects in Rust and other popular languages. My first post, as an example, is about implementing a reusable GitHub action in both Rust and TypeScript (one of the most popular methods).

Here are some other project types and languages I'm comfortable with as ideas. Please comment with a combo (even if the components are not listed!) that you would like to see compared with Rust.

## Potential projects

- A CLI tool
- A GUI tool
- A web server (GraphQL or OpenAPI)
- A bot (e.g., GitHub or Discord)

Intentionally not on this list are things I anticipate being significantly harder in Rust: mobile apps and browser UIs. Even GUIs I'm unsure about.

## Languages I'm comfortable with

Languages outside of this list are not out of the question, but they'd probably give Rust an unfair advantage.

- Python
- Go
- TypeScript
- JavaScript
- Kotlin

## An example

> I'd really like to see a comparison with a CLI built in Go, since that seems to be a really popular choice for CLIs these days.

Please comment your ideas below!

**Comment by CorinneGedmanNj** [comment]
Here are some potential combinations:

A CLI tool: Rust vs Python
A GUI tool: Rust vs TypeScript
A web server (GraphQL or OpenAPI): Rust vs Go
A bot: Rust vs JavaScript
Hope this helps!

**Reply by dbanty** [reply]
Thanks for your suggestions!!

**Comment by TyJephson7B** [comment]
Comparing the implementation of a CLI tool in Rust and Python would be an interesting comparison, as both languages are commonly used for command line interfaces and offer different advantages in terms of performance and ease of use.

**Reply by dbanty** [reply]
That's two for Rust vs Python when building a CLI—excellent!

**Comment by come-raczy** [comment]
Perhaps you could cover one of the most common things that many programmers do: processing strings.

I know, in the context of Rust, this might sound like a slightly ambiguous suggestion. What strings am I talking about? std::str? std::string::String? std::ffi::CString? std::ffi::OsString? std::os::unix::ffi::OsStringExt? std::os::windows::ffi::OsStringExt? Any of their borrowed representation e.g: OsStr, CStr? Or are we talking about many related but fully incompatible representations like the various representations of vectors of bytes and other characters of various widths and encodings? 

I'll let you choose and make the case that "Rust is not hard" when it comes to string processing (including various aspects of IOs, usage across multiple libraries, iteration over non latin character sets, etc.)
